src/index.js

```diff
@@ -1,31 +1,83 @@
 import './style.css';
 
-const tasks = [
-  {
-    description: 'wash the dishes',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'buy the house',
-    completed: false,
-    index: 2,
-  },
-];
-
-let i = 0;
-
-const listStorage = document.querySelector('.list-storage');
-tasks.forEach((task) => {
-  const toDoList = document.createElement('div');
-  toDoList.classList.add('item', `item-${i}`);
-  toDoList.innerHTML = `
-  <div class="label">
-    <input type="checkbox" class="checkbox">
-    <input type="text" name="name" value="${task.description}" readonly class="text-center">
-   </div>
-   <i class="fa-solid fa-trash-can icon-3"></i>`;
-
-  listStorage.appendChild(toDoList);
-  i += 1;
-});
\ No newline at end of file
+const taskInput = document.querySelector('.list-item');
+const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+// Function to render tasks in the DOM
+function renderTasks() {
+  const listStorage = document.querySelector('.list-storage');
+  listStorage.innerHTML = '';
+
+  tasks.forEach((task, index) => {
+    const toDoList = document.createElement('div');
+    toDoList.classList.add('item', `item-${index}`);
+    toDoList.innerHTML = `
+      <div class="label">
+        <input type="checkbox" class="checkbox">
+        <input type="text" name="name" value="${task.description}" readonly class="text-center">
+      </div>
+      <i class="fa-solid fa-trash-can icon-3 btn-${index}" data-index="${index}"></i>`;
+
+    listStorage.appendChild(toDoList);
+  });
+
+  // Attach event listeners to remove buttons
+  const removeButtons = document.querySelectorAll('.icon-3');
+  removeButtons.forEach((removeButton, index) => {
+    // eslint-disable-next-line no-use-before-define
+    removeButton.addEventListener('click', () => removeTask(index));
+  });
+
+  // Attach event listeners to text inputs for editing
+  const textCenterInputs = document.querySelectorAll('.text-center');
+  textCenterInputs.forEach((input) => {
+    input.addEventListener('click', () => {
+      if (input.hasAttribute('readonly')) {
+        input.removeAttribute('readonly');
+      } else {
+        input.setAttribute('readonly', 'true');
+      }
+    });
+  });
+}
+
+// Use info from input to add to list
+const addTask = () => {
+  if (taskInput.value) {
+    const newIndex = tasks.length + 1;
+    tasks.push({
+      description: taskInput.value,
+      completed: false,
+      index: newIndex,
+    });
+  }
+
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  taskInput.value = '';
+
+  renderTasks(); // Update the DOM
+
+  return tasks;
+};
```

Comment:
You did a good job working on this milestone! 💯 However, there are still some requirements that are missing to mark the project as approved:

According to the requirements:
![image](https://github.com/2004-okari/TODOLIST/assets/77069011/61e2754c-b436-4d09-9687-c255987bab27)

- [ ] Please consider moving the new logic (create, edit, remove) to a new file. 🤓 🔧 
- [ ] You did great working on your add and remove functions, they work as expected! 💯 However, I see that you added a listener to "simulate" the editing of a task, and, although you can modify the content of the input, the value isn't persisted in the local storage. Also, I suggest that you create a separate function `editTask` that holds the logic instead of writing it directly in the event listener! 🤓 🔧